### ISSUE TITLE
fatal: unsafe repository REPO is owned by someone else

### ISSUE THREAD

**Issue by sarasensible** [devops-infra/action-pull-request#78]
## :memo:  fatal: unsafe repository ('/github/workspace' is owned by someone else) 

Same issue as https://github.com/gaurav-nelson/github-action-markdown-link-check/issues/132 . 
From the recent commit to master it looks like you guys are already aware of this, but the fix doesn't seem to be working yet.

## :warning: Checklist
* [x] Provided a clear and concise description of what the issue is.
* [x] Given a clear and concise description of what is expected.
* [x] Proposed a clear and concise description of any alternative solutions or other features.
* [x] Added any other context or screenshots about the feature request.
* [ ] Associated pull request has been already created and link was provided.

*Check [CONTRIBUTING.md][contributing] and [CODE_OF_CONDUCT.md][code] for more information*

[contributing]: https://github.com/devops-infra/.github/blob/master/CONTRIBUTING.md
[code]: https://github.com/devops-infra/.github/blob/master/CODE_OF_CONDUCT.md

**Comment by sarasensible** [comment]
Looks like the issue is the git version - set to `2.34.2` in the Dockerfile, but only `2.35.3` supports the safe directory syntax.   Going to try forking this and attempt it myself.

**Comment by ChristophShyper** [comment]
Hello. 
I'm aware, that today's weekly update bumped `git` version to the broken one. Apparently, it has been already fixed in a newer version, but it hasn't been updated in Alpine's packages repo yet. 
They've never made breaking changes before... I'll change my workflow to prevent in future.
For now, I've tried a couple of workarounds but none of them worked, unfortunately.
Hopefully, it'll be fixed soon.

**Comment by sarasensible** [comment]
Hi @ChristophShyper this does seem to be a drastic response to a security vulnerability and it looks like it's breaking solutions like yours across the interwebs.  

Is there a reason upgrading the git version to 2.35.3 won't work in this case?

**Comment by sarasensible** [comment]
Never mind, I tried it for myself and I get this error:

```
#7 2.392 ERROR: unable to select packages:
#7 2.421   git-2.34.2-r0:
#7 2.421     breaks: world[git~2.35.3]
------
executor failed running [/bin/sh -c set -eux ;  chmod +x /entrypoint.sh ;  apk update --no-cache ;  apk add --no-cache -X http://dl-cdn.alpinelinux.org/alpine/edge/testing hub~=2.14.2 ;  apk add --no-cache     bash~=5.1.16     git~=2.35.3     jq~=1.6 ;  rm -rf /var/cache/* ;  rm -rf /root/.cache/*]: exit code: 1
```

I see this is with the latest alpine too.

**Comment by sarasensible** [comment]
Looks like
```
export GIT_CEILING_DIRECTORIES=/__w
```
might be an alternative.  Sourced from https://github.com/actions/checkout/issues/760#issuecomment-1102556218 and  https://nvd.nist.gov/vuln/detail/CVE-2022-24765

**Comment by sarasensible** [comment]
I tried using `export GIT_CEILING_DIRECTORIES=/github` in a fork and it worked for me!  I'll open up a PR from my fork.

**Comment by sarasensible** [comment]
Update:  I was not able to get the above to work.  Trying to see if any other base image would work instead of alpine.

**Comment by ChristophShyper** [comment]
As I was saying the newest available version is 2.34.2, see https://pkgs.alpinelinux.org/packages?name=git&branch=v3.15

**Comment by ChristophShyper** [comment]
After some trials I've restored previous functionality.
Now it's using Ubuntu instead of Alpine, because only in that way I can add `git-core/ppa` with the newest git packages.
